dnsmasq: Delay init to avoid race conditions

diizzyy opened the pull request:
Delay startup of dnsmasq so interfaces have time to initialize
Fixes FS#1542 (and yes, it's ugly) :-)

Signed-off-by: Daniel Engberg <daniel.engberg.lists@pyret.net>

diizzyy committed:
dnsmasq: Delay init to avoid race conditions

Delay startup of dnsmasq so interfaces have time to initialize
Fixes FS#1542

Signed-off-by: Daniel Engberg <daniel.engberg.lists@pyret.net>

dedeckeh commented:
This really looks ugly; we should try to find a better solution for the observed race condition

diizzyy commented:
```
Mon Jun 18 07:00:59 2018 user.notice : Added device handler type: tunnel
Mon Jun 18 07:00:59 2018 user.notice : Added device handler type: Network device
Mon Jun 18 07:00:59 2018 user.notice : Added device handler type: bridge
Mon Jun 18 07:00:59 2018 user.notice : Added device handler type: veth
Mon Jun 18 07:00:59 2018 user.notice : Added device handler type: macvlan
Mon Jun 18 07:00:59 2018 user.notice : Added device handler type: 8021ad
Mon Jun 18 07:00:59 2018 user.notice : Added device handler type: 8021q
Mon Jun 18 07:01:00 2018 kern.info kernel: [   26.682623] br-lan: port 1(eth0) entered blocking state
Mon Jun 18 07:01:00 2018 kern.info kernel: [   26.687918] br-lan: port 1(eth0) entered disabled state
Mon Jun 18 07:01:00 2018 kern.info kernel: [   26.693591] device eth0 entered promiscuous mode
Mon Jun 18 07:01:00 2018 kern.info kernel: [   26.715699] br-lan: port 1(eth0) entered blocking state
Mon Jun 18 07:01:00 2018 kern.info kernel: [   26.720996] br-lan: port 1(eth0) entered forwarding state
Mon Jun 18 07:01:00 2018 kern.info kernel: [   26.726782] IPv6: ADDRCONF(NETDEV_UP): br-lan: link is not ready
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Interface 'lan' is enabled
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Interface 'lan' is setting up now
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Interface 'lan' is now up
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Interface 'loopback' is enabled
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Interface 'loopback' is setting up now
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Interface 'loopback' is now up
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Interface 'wan' is enabled
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Interface 'wan6' is enabled
Mon Jun 18 07:01:00 2018 daemon.notice netifd: bridge 'br-lan' link is up
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Interface 'lan' has link connectivity
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Network device 'lo' link is up
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Interface 'loopback' has link connectivity
Mon Jun 18 07:01:00 2018 kern.info kernel: [   26.751100] IPv6: ADDRCONF(NETDEV_UP): eth1: link is not ready
Mon Jun 18 07:01:00 2018 authpriv.info dropbear[868]: Not backgrounding
Mon Jun 18 07:01:00 2018 user.notice firewall: Reloading firewall due to ifup of lan (br-lan)
Mon Jun 18 07:01:00 2018 daemon.notice netifd: bridge 'br-lan' link is down
Mon Jun 18 07:01:00 2018 daemon.notice netifd: Interface 'lan' has link connectivity loss
Mon Jun 18 07:01:00 2018 kern.info kernel: [   27.649400] br-lan: port 1(eth0) entered disabled state
Mon Jun 18 07:01:01 2018 daemon.info procd: - init complete -
Mon Jun 18 07:01:04 2018 kern.notice kernel: [   30.753079] eth0: 1000 Mbps Full duplex, port 0, queue 0
Mon Jun 18 07:01:04 2018 kern.info kernel: [   30.758490] br-lan: port 1(eth0) entered blocking state
Mon Jun 18 07:01:04 2018 kern.info kernel: [   30.763771] br-lan: port 1(eth0) entered forwarding state
Mon Jun 18 07:01:04 2018 daemon.notice netifd: Network device 'eth0' link is up
Mon Jun 18 07:01:04 2018 kern.info kernel: [   30.769481] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
Mon Jun 18 07:01:04 2018 daemon.notice netifd: bridge 'br-lan' link is up
Mon Jun 18 07:01:04 2018 daemon.notice netifd: Interface 'lan' has link connectivity
Mon Jun 18 07:01:06 2018 user.notice dnsmasq: DNS rebinding protection is active, will discard upstream RFC1918 responses!
Mon Jun 18 07:01:06 2018 user.notice dnsmasq: Allowing 127.0.0.0/8 responses
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: started, version 2.80test2 cachesize 150
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: DNS service limited to local subnets
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: compile time options: IPv6 GNU-getopt no-DBus no-i18n no-IDN DHCP no-DHCPv6 no-Lua TFTP no-conntrack no-ipset no-auth no-DNSSEC no-ID loop-detect inotify dumpfile
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq-dhcp[1107]: DHCP, IP range 192.168.1.100 -- 192.168.1.249, lease time 12h
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: using local addresses only for domain test
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: using local addresses only for domain onion
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: using local addresses only for domain localhost
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: using local addresses only for domain local
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: using local addresses only for domain invalid
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: using local addresses only for domain bind
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: using local addresses only for domain lan
Mon Jun 18 07:01:09 2018 daemon.warn dnsmasq[1107]: no servers found in /tmp/resolv.conf.auto, will retry
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: read /etc/hosts - 4 addresses
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: read /tmp/hosts/dhcp.cfg01411c - 1 addresses
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq-dhcp[1107]: read /etc/ethers - 0 addresses
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: read /etc/hosts - 4 addresses
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq[1107]: read /tmp/hosts/dhcp.cfg01411c - 1 addresses
Mon Jun 18 07:01:09 2018 daemon.info dnsmasq-dhcp[1107]: read /etc/ethers - 0 addresses
```

If someone has an better idea I'm all ears

diizzyy commented:
@dedeckeh 
Yes, it's hackish but I'm unaware on how to solve this nicely :-/

blogic commented:
no way. can we use interface triggers here ?

dedeckeh commented:
@blogic dnsmasq init script already uses interface triggers to trigger dnsmasq reload in case an interface gets active
@diizzyy the root cause of the problem needs to be understood; is the procd interface trigger not in place or is dnsmasq reload not triggered by procd or ... ?

diizzyy commented:
@dedeckeh 
I'm not sure exactly what's going on but there's quite a bit of link state changes that may be causing something funny.

```
Serial log:

[    0.000000] Linux version 4.14.50 (buildbot@crazyhorse) (gcc version 7.3.0 (OpenWrt GCC 7.3.0 r7288-a9222e0)) #0 SMP Wed Jun 20 13:42:22 2018
[    0.000000] CVMSEG size: 2 cache lines (256 bytes)
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 000d0601 (Cavium Octeon+)
[    0.000000] Checking for the multiply/shift bug... no.
[    0.000000] Checking for the daddiu bug... no.
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 0000000005400000 @ 0000000002a00000 (usable)
[    0.000000]  memory: 0000000007c00000 @ 0000000008200000 (usable)
[    0.000000]  memory: 000000000fc00000 @ 0000000410000000 (usable)
[    0.000000]  memory: 0000000001869a60 @ 0000000001100000 (usable)
[    0.000000] Wasting 243712 bytes for tracking 4352 unused pages
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Using internal Device Tree.
[    0.000000] software IO TLB [mem 0x0398e000-0x0798e000] (64MB) mapped at [800000000398e000-800000000798dfff]
[    0.000000] Primary instruction cache 32kB, virtually tagged, 4 way, 64 sets, linesize 128 bytes.
[    0.000000] Primary data cache 16kB, 64-way, 2 sets, linesize 128 bytes.
[    0.000000] Zone ranges:
[    0.000000]   DMA32    [mem 0x0000000000000000-0x00000000efffffff]
[    0.000000]   Normal   [mem 0x00000000f0000000-0x000000041fbfffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000002968fff]
[    0.000000]   node   0: [mem 0x0000000002a00000-0x0000000007dfffff]
[    0.000000]   node   0: [mem 0x0000000008200000-0x000000000fdfffff]
[    0.000000]   node   0: [mem 0x0000000410000000-0x000000041fbfffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x000000041fbfffff]
[    0.000000] random: get_random_bytes called from start_kernel+0x90/0x504 with crng_init=0
[    0.000000] percpu: Embedded 19 pages/cpu @80000000079f9000 s39472 r8192 d30160 u77824
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 126606
[    0.000000] Kernel command line: mtdparts=phys_mapped_flash:512k(boot0)ro,512k(boot1)ro,64k(eeprom)ro root=/dev/sda2 rootfstype=squashfs,ext4 rootwait  console=ttyS0,115200
[    0.000000] PID hash table entries: 2048 (order: 2, 16384 bytes)
[    0.000000] Dentry cache hash table entries: 65536 (order: 7, 524288 bytes)
[    0.000000] Inode-cache hash table entries: 32768 (order: 6, 262144 bytes)
[    0.000000] Memory: 396916K/513444K available (5292K kernel code, 347K rwdata, 1292K rodata, 1304K init, 16742K bss, 116528K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=128, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] Hierarchical RCU implementation.
[    0.000000]  CONFIG_RCU_FANOUT set to non-default value of 32
[    0.000000]  RCU restricting CPUs from NR_CPUS=16 to nr_cpu_ids=2.
[    0.000000] RCU: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=2
[    0.000000] NR_IRQS: 127
[    7.556406] clocksource: OCTEON_CVMCOUNT: mask: 0xffffffffffffffff max_cycles: 0xe6a171a037, max_idle_ns: 881590485102 ns
[    7.567275] Calibrating delay loop (skipped) preset value.. 1000.00 BogoMIPS (lpj=2000000)
[    7.575474] pid_max: default: 32768 minimum: 301
[    7.580211] Mount-cache hash table entries: 1024 (order: 1, 8192 bytes)
[    7.586694] Mountpoint-cache hash table entries: 1024 (order: 1, 8192 bytes)
[    7.594412] Checking for the daddi bug... no.
[    7.599665] Hierarchical SRCU implementation.
[    7.604698] smp: Bringing up secondary CPUs ...
[    7.609758] SMP: Booting CPU01 (CoreId  1)...
[    7.614146] CPU1 revision is: 000d0601 (Cavium Octeon+)
[    7.614367] smp: Brought up 1 node, 2 CPUs
[    7.627584] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    7.637224] futex hash table entries: 512 (order: 4, 65536 bytes)
[    7.644068] NET: Registered protocol family 16
[    7.658985] Not in host mode, PCI Controller not initialized
[    7.680267] SCSI subsystem initialized
[    7.684454] usbcore: registered new interface driver usbfs
[    7.689990] usbcore: registered new interface driver hub
[    7.695329] usbcore: registered new device driver usb
[    7.702792] clocksource: Switched to clocksource OCTEON_CVMCOUNT
[    7.710100] NET: Registered protocol family 2
[    7.715499] TCP established hash table entries: 4096 (order: 3, 32768 bytes)
[    7.722530] TCP bind hash table entries: 4096 (order: 5, 131072 bytes)
[    7.729083] TCP: Hash tables configured (established 4096 bind 4096)
[    7.735468] UDP hash table entries: 256 (order: 2, 24576 bytes)
[    7.741333] UDP-Lite hash table entries: 256 (order: 2, 24576 bytes)
[    7.747926] NET: Registered protocol family 1
[    7.753926] Crashlog allocated RAM at address 0x3f00000
[    7.759482] workingset: timestamp_bits=62 max_order=17 bucket_order=0
[    7.777976] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    7.784098] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    7.799341] io scheduler noop registered
[    7.803139] io scheduler deadline registered (default)
[    7.808849] octeon_gpio 1070000000800.gpio-controller: OCTEON GPIO driver probed.
[    7.816674] Serial: 8250/16550 driver, 16 ports, IRQ sharing enabled
[    7.827060] console [ttyS0] disabled
[    7.830534] 1180000000800.serial: ttyS0 at MMIO 0x1180000000800 (irq = 41, base_baud = 31250000) is a OCTEON
[    7.840346] console [ttyS0] enabled
[    7.840346] console [ttyS0] enabled
[    7.847255] bootconsole [early0] disabled
[    7.847255] bootconsole [early0] disabled
[    7.855944] 1180000000c00.serial: ttyS1 at MMIO 0x1180000000c00 (irq = 42, base_baud = 31250000) is a OCTEON
[    7.866426] octeon_rng octeon_rng: Octeon Random Number Generator
[    7.872915] cacheinfo: Failed to find cpu0 device node
[    7.878181] cacheinfo: Unable to detect cache hierarchy for CPU 0
[    7.891156] loop: module loaded
[    7.895234] of-flash 1f400000.nor: Can't get bank width from device tree
[    7.902161] random: fast init done
[    7.902381] libphy: mdio_octeon: probed
[    7.902420] [Firmware Warn]: /soc@0/mdio@1180000001800/ethernet-phy@7: Whitelisted compatible string. Please remove
[    7.920637] [Firmware Warn]: /soc@0/mdio@1180000001800/ethernet-phy@6: Whitelisted compatible string. Please remove
[    7.931781] [Firmware Warn]: /soc@0/mdio@1180000001800/ethernet-phy@5: Whitelisted compatible string. Please remove
[    7.942966] mdio_octeon 1180000001800.mdio: Probed
[    7.948388] libphy: Fixed MDIO Bus: probed
[    7.952749] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    7.959322] ehci-pci: EHCI PCI platform driver
[    7.963911] ehci-platform: EHCI generic platform driver
[    7.969418] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
[    7.975699] ohci-platform: OHCI generic platform driver
[    7.981310] usbcore: registered new interface driver usb-storage
[    7.987419] octeon_wdt: Initial granularity 5 Sec
[    7.996199] Interface 0 has 3 ports (RGMII)
[    8.004703] octeon-hcd 16f0010000000.usbc: Octeon Host Controller
[    8.010920] octeon-hcd 16f0010000000.usbc: new USB bus registered, assigned bus number 1
[    8.019145] octeon-hcd 16f0010000000.usbc: irq 56, io mem 0x00000000
[    8.026551] hub 1-0:1.0: USB hub found
[    8.030469] hub 1-0:1.0: 1 port detected
[    8.035184] octeon-hcd 16f0010000000.usbc: Registered HCD for port 0 on irq 56
[    8.043905] NET: Registered protocol family 10
[    8.050409] Segment Routing with IPv6
[    8.054257] NET: Registered protocol family 17
[    8.058841] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
[    8.071844] 8021q: 802.1Q VLAN Support v1.8
[    8.076342] Bootbus flash: Setting flash for 4MB flash at 0x1f800000
[    8.082861] phys_mapped_flash: Found 1 x16 devices at 0x0 in 8-bit bank. Manufacturer ID 0x0000c2 Chip ID 0x0000a7
[    8.093245] Amd/Fujitsu Extended Query Table at 0x0040
[    8.098443]   Amd/Fujitsu Extended Query version 1.1.
[    8.103523] phys_mapped_flash: Swapping erase regions for top-boot CFI table.
[    8.110690] number of CFI chips: 1
[    8.114123] 3 cmdlinepart partitions found on MTD device phys_mapped_flash
[    8.121025] Creating 3 MTD partitions on "phys_mapped_flash":
[    8.126804] 0x000000000000-0x000000080000 : "boot0"
[    8.132853] 0x000000080000-0x000000100000 : "boot1"
[    8.138977] 0x000000100000-0x000000110000 : "eeprom"
[    8.147227] OF: fdt: not creating '/sys/firmware/fdt': CRC check failed
[    8.154450] Waiting for root device /dev/sda2...
[    8.434726] usb 1-1: new high-speed USB device number 2 using octeon-hcd
[    8.909723] usb-storage 1-1:1.0: USB Mass Storage device detected
[    8.916409] scsi host0: usb-storage 1-1:1.0
[    9.939693] scsi 0:0:0:0: Direct-Access     UFD 3.0  Silicon-Power8G  PMAP PQ: 0 ANSI: 6
[   10.232728] sd 0:0:0:0: [sda] 15466496 512-byte logical blocks: (7.92 GB/7.38 GiB)
[   10.241761] sd 0:0:0:0: [sda] Write Protect is off
[   10.247947] sd 0:0:0:0: [sda] No Caching mode page found
[   10.253348] sd 0:0:0:0: [sda] Assuming drive cache: write through
[   10.264948]  sda: sda1 sda2
[   10.272391] sd 0:0:0:0: [sda] Attached SCSI removable disk
[   10.282278] VFS: Mounted root (squashfs filesystem) readonly on device 8:2.
[   10.290446] Freeing unused kernel memory: 1304K
[   10.295036] This architecture does not have kernel memory protection.
[   10.735488] init: Console is alive
[   10.739335] init: - watchdog -
[   10.993648] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[   11.001445] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[   11.017022] init: - preinit -
[   11.554002] random: jshn: uninitialized urandom read (4 bytes read)
[   11.603066] random: jshn: uninitialized urandom read (4 bytes read)
[   11.642855] random: jshn: uninitialized urandom read (4 bytes read)
[   11.680144] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[   15.157139] F2FS-fs (loop0): Mounted with checkpoint version = 8f5dc8c
[   15.165898] mount_root: switching to f2fs overlay
[   15.766874] eth0: 1000 Mbps Full duplex, port 0, queue 0
[   15.772292] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   15.792823] urandom-seed: Seeding with /etc/urandom.seed
[   15.892092] eth0: Link down
[   15.905493] procd: - early -
[   15.908602] procd: - watchdog -
[   16.587588] procd: - watchdog -
[   16.591301] procd: - ubus -
[   16.643534] urandom_read: 1 callbacks suppressed
[   16.643542] random: ubusd: uninitialized urandom read (4 bytes read)
[   16.692112] random: ubusd: uninitialized urandom read (4 bytes read)
[   16.699071] random: ubusd: uninitialized urandom read (4 bytes read)
[   16.706963] procd: - init -
Please press Enter to activate this console.
[   16.966655] kmodloader: loading kernel modules from /etc/modules.d/*
[   16.978376] ip6_tables: (C) 2000-2006 Netfilter Core Team
[   17.122949] ip_tables: (C) 2000-2006 Netfilter Core Team
[   17.263868] nf_conntrack version 0.5.0 (3584 buckets, 14336 max)
[   17.387780] xt_time: kernel timezone is -0000
[   17.459694] PPP generic driver version 2.4.2
[   17.465444] NET: Registered protocol family 24
[   17.473113] kmodloader: done loading kernel modules from /etc/modules.d/*
[   18.774757] random: crng init done
[   18.778177] random: 3 urandom warning(s) missed due to ratelimiting
[   22.483943] br-lan: port 1(eth0) entered blocking state
[   22.489247] br-lan: port 1(eth0) entered disabled state
[   22.494983] device eth0 entered promiscuous mode
[   22.502774] br-lan: port 1(eth0) entered blocking state
[   22.508060] br-lan: port 1(eth0) entered forwarding state
[   22.513778] IPv6: ADDRCONF(NETDEV_UP): br-lan: link is not ready
[   22.551044] IPv6: ADDRCONF(NETDEV_UP): eth1: link is not ready
[   23.507449] br-lan: port 1(eth0) entered disabled state
[   26.578823] eth0: 1000 Mbps Full duplex, port 0, queue 0
[   26.584222] br-lan: port 1(eth0) entered blocking state
[   26.589496] br-lan: port 1(eth0) entered forwarding state
[   26.595495] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready

Output from logread

Wed Jun 20 13:42:38 2018 kern.notice kernel: [    0.000000] Linux version 4.14.50 (buildbot@crazyhorse) (gcc version 7.3.0 (OpenWrt GCC 7.3.0 r7288-a9222e0)) #0 SMP Wed Jun 20 13:42:22 2018
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    0.000000] CVMSEG size: 2 cache lines (256 bytes)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] bootconsole [early0] enabled
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] CPU0 revision is: 000d0601 (Cavium Octeon+)
Wed Jun 20 13:42:38 2018 kern.warn kernel: [    0.000000] Checking for the multiply/shift bug... no.
Wed Jun 20 13:42:38 2018 kern.warn kernel: [    0.000000] Checking for the daddiu bug... no.
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Determined physical RAM map:
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]  memory: 0000000005400000 @ 0000000002a00000 (usable)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]  memory: 0000000007c00000 @ 0000000008200000 (usable)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]  memory: 000000000fc00000 @ 0000000410000000 (usable)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]  memory: 0000000001869a60 @ 0000000001100000 (usable)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Wasting 243712 bytes for tracking 4352 unused pages
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Initrd not found or empty - disabling initrd
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Using internal Device Tree.
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] software IO TLB [mem 0x0398e000-0x0798e000] (64MB) mapped at [800000000398e000-800000000798dfff]
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    0.000000] Primary instruction cache 32kB, virtually tagged, 4 way, 64 sets, linesize 128 bytes.
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    0.000000] Primary data cache 16kB, 64-way, 2 sets, linesize 128 bytes.
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Zone ranges:
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]   DMA32    [mem 0x0000000000000000-0x00000000efffffff]
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]   Normal   [mem 0x00000000f0000000-0x000000041fbfffff]
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Movable zone start for each node
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Early memory node ranges
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]   node   0: [mem 0x0000000000000000-0x0000000002968fff]
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]   node   0: [mem 0x0000000002a00000-0x0000000007dfffff]
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]   node   0: [mem 0x0000000008200000-0x000000000fdfffff]
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]   node   0: [mem 0x0000000410000000-0x000000041fbfffff]
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x000000041fbfffff]
Wed Jun 20 13:42:38 2018 kern.debug kernel: [    0.000000] On node 0 totalpages: 128361
Wed Jun 20 13:42:38 2018 kern.debug kernel: [    0.000000]   DMA32 zone: 873 pages used for memmap
Wed Jun 20 13:42:38 2018 kern.debug kernel: [    0.000000]   DMA32 zone: 0 pages reserved
Wed Jun 20 13:42:38 2018 kern.debug kernel: [    0.000000]   DMA32 zone: 63849 pages, LIFO batch:15
Wed Jun 20 13:42:38 2018 kern.debug kernel: [    0.000000]   Normal zone: 882 pages used for memmap
Wed Jun 20 13:42:38 2018 kern.debug kernel: [    0.000000]   Normal zone: 64512 pages, LIFO batch:15
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    0.000000] random: get_random_bytes called from start_kernel+0x90/0x504 with crng_init=0
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] percpu: Embedded 19 pages/cpu @80000000079f9000 s39472 r8192 d30160 u77824
Wed Jun 20 13:42:38 2018 kern.debug kernel: [    0.000000] pcpu-alloc: s39472 r8192 d30160 u77824 alloc=19*4096
Wed Jun 20 13:42:38 2018 kern.debug kernel: [    0.000000] pcpu-alloc: [0] 0 [0] 1
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 126606
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    0.000000] Kernel command line: mtdparts=phys_mapped_flash:512k(boot0)ro,512k(boot1)ro,64k(eeprom)ro root=/dev/sda2 rootfstype=squashfs,ext4 rootwait  console=ttyS0,115200
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] PID hash table entries: 2048 (order: 2, 16384 bytes)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Dentry cache hash table entries: 65536 (order: 7, 524288 bytes)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Inode-cache hash table entries: 32768 (order: 6, 262144 bytes)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Memory: 396916K/513444K available (5292K kernel code, 347K rwdata, 1292K rodata, 1304K init, 16742K bss, 116528K reserved, 0K cma-reserved)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] SLUB: HWalign=128, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] Hierarchical RCU implementation.
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]       CONFIG_RCU_FANOUT set to non-default value of 32
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000]       RCU restricting CPUs from NR_CPUS=16 to nr_cpu_ids=2.
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] RCU: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=2
Wed Jun 20 13:42:38 2018 kern.info kernel: [    0.000000] NR_IRQS: 127
Wed Jun 20 13:42:38 2018 kern.info kernel: [   7.556406] clocksource: OCTEON_CVMCOUNT: mask: 0xffffffffffffffff max_cycles: 0xe6a171a037, max_idle_ns: 881590485102 ns
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.567275] Calibrating delay loop (skipped) preset value.. 1000.00 BogoMIPS (lpj=2000000)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.575474] pid_max: default: 32768 minimum: 301
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.580211] Mount-cache hash table entries: 1024 (order: 1, 8192 bytes)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.586694] Mountpoint-cache hash table entries: 1024 (order: 1, 8192 bytes)
Wed Jun 20 13:42:38 2018 kern.warn kernel: [    7.594412] Checking for the daddi bug... no.
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.599665] Hierarchical SRCU implementation.
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.604698] smp: Bringing up secondary CPUs ...
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.609758] SMP: Booting CPU01 (CoreId  1)...
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.614146] CPU1 revision is: 000d0601 (Cavium Octeon+)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.614367] smp: Brought up 1 node, 2 CPUs
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.627584] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.637224] futex hash table entries: 512 (order: 4, 65536 bytes)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.644068] NET: Registered protocol family 16
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    7.658985] Not in host mode, PCI Controller not initialized
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    7.680267] SCSI subsystem initialized
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.684454] usbcore: registered new interface driver usbfs
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.689990] usbcore: registered new interface driver hub
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.695329] usbcore: registered new device driver usb
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.702792] clocksource: Switched to clocksource OCTEON_CVMCOUNT
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.710100] NET: Registered protocol family 2
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.715499] TCP established hash table entries: 4096 (order: 3, 32768 bytes)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.722530] TCP bind hash table entries: 4096 (order: 5, 131072 bytes)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.729083] TCP: Hash tables configured (established 4096 bind 4096)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.735468] UDP hash table entries: 256 (order: 2, 24576 bytes)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.741333] UDP-Lite hash table entries: 256 (order: 2, 24576 bytes)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.747926] NET: Registered protocol family 1
Wed Jun 20 13:42:38 2018 kern.debug kernel: [    7.752243] PCI: CLS 0 bytes, default 128
Wed Jun 20 13:42:38 2018 kern.warn kernel: [    7.753926] Crashlog allocated RAM at address 0x3f00000
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.759482] workingset: timestamp_bits=62 max_order=17 bucket_order=0
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.777976] squashfs: version 4.0 (2009/01/31) Phillip Lougher
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.784098] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.799341] io scheduler noop registered
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.803139] io scheduler deadline registered (default)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.808849] octeon_gpio 1070000000800.gpio-controller: OCTEON GPIO driver probed.
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.816674] Serial: 8250/16550 driver, 16 ports, IRQ sharing enabled
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.827060] console [ttyS0] disabled
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.830534] 1180000000800.serial: ttyS0 at MMIO 0x1180000000800 (irq = 41, base_baud = 31250000) is a OCTEON
Wed Jun 20 13:42:38 2018 kern.info kernel: [    6] console [ttyS0] enabled
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.847255] bootconsole [early0] disabled
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.855944] 1180000000c00.serial: ttyS1 at MMIO 0x1180000000c00 (irq = 42, base_baud = 31250000) is a OCTEON
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.866426] octeon_rng octeon_rng: Octeon Random Number Generator
Wed Jun 20 13:42:38 2018 kern.err kernel: [    7.872915] cacheinfo: Failed to find cpu0 device node
Wed Jun 20 13:42:38 2018 kern.warn kernel: [    7.878181] cacheinfo: Unable to detect cache hierarchy for CPU 0
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.891156] loop: module loaded
Wed Jun 20 13:42:38 2018 kern.err kernel: [    7.895234] of-flash 1f400000.nor: Can't get bank width from device tree
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    7.902161] random: fast init done
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.902381] libphy: mdio_octeon: probed
Wed Jun 20 13:42:38 2018 kern.warn kernel: [    7.902420] [Firmware Warn]: /soc@0/mdio@1180000001800/ethernet-phy@7: Whitelisted compatible string. Please remove
Wed Jun 20 13:42:38 2018 kern.warn kernel: [    7.920637] [Firmware Warn]: /soc@0/mdio@1180000001800/ethernet-phy@6: Whitelisted compatible string. Please remove
Wed Jun 20 13:42:38 2018 kern.warn kernel: [    7.931781] [Firmware Warn]: /soc@0/mdio@1180000001800/ethernet-phy@5: Whitelisted compatible string. Please remove
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.942966] mdio_octeon 1180000001800.mdio: Probed
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.948388] libphy: Fixed MDIO Bus: probed
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.952749] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.959322] ehci-pci: EHCI PCI platform driver
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.963911] ehci-platform: EHCI generic platform driver
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.969418] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.975699] ohci-platform: OHCI generic platform driver
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.981310] usbcore: registered new interface driver usb-storage
Wed Jun 20 13:42:38 2018 kern.info kernel: [    7.987419] octeon_wdt: Initial granularity 5 Sec
Wed Jun 20 13:42:38 2018 kern.warn kernel: [    7.996199] Interface 0 has 3 ports (RGMII)
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.004703] octeon-hcd 16f0010000000.usbc: Octeon Host Controller
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.010920] octeon-hcd 16f0010000000.usbc: new USB bus registered, assigned bus number 1
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.019145] octeon-hcd 16f0010000000.usbc: irq 56, io mem 0x00000000
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.026551] hub 1-0:1.0: USB hub found
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.030469] hub 1-0:1.0: 1 port detected
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.035184] octeon-hcd 16f0010000000.usbc: Registered HCD for port 0 on irq 56
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.043905] NET: Registered protocol family 10
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.050409] Segment Routing with IPv6
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.054257] NET: Registered protocol family 17
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.058841] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.071844] 8021q: 802.1Q VLAN Support v1.8
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    8.076342] Bootbus flash: Setting flash for 4MB flash at 0x1f800000
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.082861] phys_mapped_flash: Found 1 x16 devices at 0x0 in 8-bit bank. Manufacturer ID 0x0000c2 Chip ID 0x0000a7
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.093245] Amd/Fujitsu Extended Query Table at 0x0040
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.098443]   Amd/Fujitsu Extended Query version 1.1.
Wed Jun 20 13:42:38 2018 kern.warn kernel: [    8.103523] phys_mapped_flash: Swapping erase regions for top-boot CFI table.
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    8.110690] number of CFI chips: 1
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    8.114123] 3 cmdlinepart partitions found on MTD device phys_mapped_flash
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    8.121025] Creating 3 MTD partitions on "phys_mapped_flash":
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    8.126804] 0x000000000000-0x000000080000 : "boot0"
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    8.132853] 0x000000080000-0x000000100000 : "boot1"
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    8.138977] 0x000000100000-0x000000110000 : "eeprom"
Wed Jun 20 13:42:38 2018 kern.warn kernel: [    8.147227] OF: fdt: not creating '/sys/firmware/fdt': CRC check failed
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.154450] Waiting for root device /dev/sda2...
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.434726] usb 1-1: new high-speed USB device number 2 using octeon-hcd
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.909723] usb-storage 1-1:1.0: USB Mass Storage device detected
Wed Jun 20 13:42:38 2018 kern.info kernel: [    8.916409] scsi host0: usb-storage 1-1:1.0
Wed Jun 20 13:42:38 2018 kern.notice kernel: [    9.939693] scsi 0:0:0:0: Direct-Access     UFD 3.0  Silicon-Power8G  PMAP PQ: 0 ANSI: 6
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   10.232728] sd 0:0:0:0: [sda] 15466496 512-byte logical blocks: (7.92 GB/7.38 GiB)
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   10.241761] sd 0:0:0:0: [sda] Write Protect is off
Wed Jun 20 13:42:38 2018 kern.debug kernel: [   10.246634] sd 0:0:0:0: [sda] Mode Sense: 23 00 00 00
Wed Jun 20 13:42:38 2018 kern.err kernel: [   10.247947] sd 0:0:0:0: [sda] No Caching mode page found
Wed Jun 20 13:42:38 2018 kern.err kernel: [   10.253348] sd 0:0:0:0: [sda] Assuming drive cache: write through
Wed Jun 20 13:42:38 2018 kern.info kernel: [   10.264948]  sda: sda1 sda2
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   10.272391] sd 0:0:0:0: [sda] Attached SCSI removable disk
Wed Jun 20 13:42:38 2018 kern.info kernel: [   10.282278] VFS: Mounted root (squashfs filesystem) readonly on device 8:2.
Wed Jun 20 13:42:38 2018 kern.info kernel: [   10.290446] Freeing unused kernel memory: 1304K
Wed Jun 20 13:42:38 2018 kern.warn kernel: [   10.295036] This architecture does not have kernel memory protection.
Wed Jun 20 13:42:38 2018 user.info kernel: [   10.735488] init: Console is alive
Wed Jun 20 13:42:38 2018 user.info kernel: [   10.739335] init: - watchdog -
Wed Jun 20 13:42:38 2018 user.info kernel: [   10.993648] kmodloader: loading kernel modules from /etc/modules-boot.d/*
Wed Jun 20 13:42:38 2018 user.info kernel: [   11.001445] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
Wed Jun 20 13:42:38 2018 user.info kernel: [   11.017022] init: - preinit -
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   11.554002] random: jshn: uninitialized urandom read (4 bytes read)
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   11.603066] random: jshn: uninitialized urandom read (4 bytes read)
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   11.642855] random: jshn: uninitialized urandom read (4 bytes read)
Wed Jun 20 13:42:38 2018 kern.info kernel: [   11.680144] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   15.157139] F2FS-fs (loop0): Mounted with checkpoint version = 8f5dc8c
Wed Jun 20 13:42:38 2018 user.info kernel: [   15.165898] mount_root: switching to f2fs overlay
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   15.766874] eth0: 1000 Mbps Full duplex, port 0, queue 0
Wed Jun 20 13:42:38 2018 kern.info kernel: [   15.772292] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Wed Jun 20 13:42:38 2018 user.warn kernel: [   15.792823] urandom-seed: Seeding with /etc/urandom.seed
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   15.892092] eth0: Link down
Wed Jun 20 13:42:38 2018 user.info kernel: [   15.905493] procd: - early -
Wed Jun 20 13:42:38 2018 user.info kernel: [   15.908602] procd: - watchdog -
Wed Jun 20 13:42:38 2018 user.info kernel: [   16.587588] procd: - watchdog -
Wed Jun 20 13:42:38 2018 user.info kernel: [   16.591301] procd: - ubus -
Wed Jun 20 13:42:38 2018 kern.warn kernel: [   16.643534] urandom_read: 1 callbacks suppressed
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   16.643542] random: ubusd: uninitialized urandom read (4 bytes read)
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   16.692112] random: ubusd: uninitialized urandom read (4 bytes read)
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   16.699071] random: ubusd: uninitialized urandom read (4 bytes read)
Wed Jun 20 13:42:38 2018 user.info kernel: [   16.706963] procd: - init -
Wed Jun 20 13:42:38 2018 user.info kernel: [   16.966655] kmodloader: loading kernel modules from /etc/modules.d/*
Wed Jun 20 13:42:38 2018 kern.info kernel: [   16.978376] ip6_tables: (C) 2000-2006 Netfilter Core Team
Wed Jun 20 13:42:38 2018 kern.info kernel: [   17.122949] ip_tables: (C) 2000-2006 Netfilter Core Team
Wed Jun 20 13:42:38 2018 kern.info kernel: [   17.263868] nf_conntrack version 0.5.0 (3584 buckets, 14336 max)
Wed Jun 20 13:42:38 2018 kern.info kernel: [   17.387780] xt_time: kernel timezone is -0000
Wed Jun 20 13:42:38 2018 kern.info kernel: [   17.459694] PPP generic driver version 2.4.2
Wed Jun 20 13:42:38 2018 kern.info kernel: [   17.465444] NET: Registered protocol family 24
Wed Jun 20 13:42:38 2018 user.info kernel: [   17.473113] kmodloader: done loading kernel modules from /etc/modules.d/*
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   18.774757] random: crng it done
Wed Jun 20 13:42:38 2018 kern.notice kernel: [   18.778177] random: 3 urandom warning(s) missed due to ratelimiting
Wed Jun 20 13:42:38 2018 user.notice dnsmasq: DNS rebinding protection is active, will discard upstream RFC1918 responses!
Wed Jun 20 13:42:39 2018 user.notice dnsmasq: Allowing 127.0.0.0/8 responses
Wed Jun 20 13:42:39 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Command failed: Not found
Wed Jun 20 13:42:39 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Failed to parse json data: unexpected end of data
Wed Jun 20 13:42:39 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Command failed: Not found
Wed Jun 20 13:42:39 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Failed to parse json data: unexpected end of data
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: started, version 2.80test2 cachesize 150
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: DNS service limited to local subnets
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: compile time options: IPv6 GNU-getopt no-DBus no-i18n no-IDN DHCP no-DHCPv6 no-Lua TFTP no-conntrack no-ipset no-auth no-DNSSEC no-ID loop-detect inotify dumpfile
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: using local addresses only for domain test
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: using local addresses only for domain onion
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: using local addresses only for domain localhost
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: using local addresses only for domain local
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: using local addresses only for domain invalid
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: using local addresses only for domain bind
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: using local addresses only for domain lan
Wed Jun 20 13:42:39 2018 daemon.warn dnsmasq[742]: no servers found in /tmp/resolv.conf.auto, will retry
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: read /etc/hosts - 4 addresses
Wed Jun 20 13:42:39 2018 daemon.info dnsmasq[742]: read /tmp/hosts/dhcp.cfg01411c - 0 addresses
Wed Jun 20 13:42:40 2018 user.notice : Added device handler type: tunnel
Wed Jun 20 13:42:40 2018 user.notice : Added device handler type: Network device
Wed Jun 20 13:42:40 2018 user.notice : Added device handler type: bridge
Wed Jun 20 13:42:40 2018 user.notice : Added device handler type: veth
Wed Jun 20 13:42:40 2018 user.notice : Added device handler type: macvlan
Wed Jun 20 13:42:40 2018 user.notice : Added device handler type: 8021ad
Wed Jun 20 13:42:40 2018 user.notice : Added device handler type: 8021q
Wed Jun 20 13:42:41 2018 kern.info kernel: [   22.483943] br-lan: port 1(eth0) entered blocking state
Wed Jun 20 13:42:41 2018 kern.info kernel: [   22.489247] br-lan: port 1(eth0) entered disabled state
Wed Jun 20 13:42:41 2018 kern.info kernel: [   22.494983] device eth0 entered promiscuous mode
Wed Jun 20 13:42:41 2018 kern.info kernel: [   22.502774] br-lan: port 1(eth0) entered blocking state
Wed Jun 20 13:42:41 nfo kernel: [   22.508060] br-lan: port 1(eth0) entered forwarding state
Wed Jun 20 13:42:41 2018 kern.info kernel: [   22.513778] IPv6: ADDRCONF(NETDEV_UP): br-lan: link is not ready
Wed Jun 20 13:42:41 2018 daemon.notice netifd: Interface 'lan' is enabled
Wed Jun 20 13:42:41 2018 daemon.notice netifd: Interface 'lan' is setting up now
Wed Jun 20 13:42:41 2018 daemon.notice netifd: Interface 'lan' is now up
Wed Jun 20 13:42:41 2018 daemon.notice netifd: Interface 'loopback' is enabled
Wed Jun 20 13:42:41 2018 daemon.notice netifd: Interface 'loopback' is setting up now
Wed Jun 20 13:42:41 2018 daemon.notice netifd: Interface 'loopback' is now up
Wed Jun 20 13:42:41 2018 kern.info kernel: [   22.551044] IPv6: ADDRCONF(NETDEV_UP): eth1: link is not ready
Wed Jun 20 13:42:41 2018 daemon.notice netifd: Interface 'wan' is enabled
Wed Jun 20 13:42:41 2018 daemon.notice netifd: Interface 'wan6' is enabled
Wed Jun 20 13:42:41 2018 daemon.notice netifd: bridge 'br-lan' link is up
Wed Jun 20 13:42:41 2018 daemon.notice netifd: Interface 'lan' has link connectivity
Wed Jun 20 13:42:41 2018 daemon.notice netifd: Network device 'lo' link is up
Wed Jun 20 13:42:41 2018 daemon.notice netifd: Interface 'loopback' has link connectivity
Wed Jun 20 13:42:41 2018 authpriv.info dropbear[891]: Not backgrounding
Wed Jun 20 13:42:42 2018 user.notice firewall: Reloading firewall due to ifup of lan (br-lan)
Wed Jun 20 13:42:42 2018 kern.info kernel: [   23.507449] br-lan: port 1(eth0) entered disabled state
Wed Jun 20 13:42:42 2018 daemon.notice netifd: bridge 'br-lan' link is down
Wed Jun 20 13:42:42 2018 daemon.notice netifd: Interface 'lan' has link connectivity loss
Wed Jun 20 13:42:42 2018 daemon.info procd: - init complete -
Wed Jun 20 13:42:43 2018 daemon.info dnsmasq[742]: read /etc/hosts - 4 addresses
Wed Jun 20 13:42:43 2018 daemon.info dnsmasq[742]: read /tmp/hosts/dhcp.cfg01411c - 0 addresses
Wed Jun 20 13:42:45 2018 kern.notice kernel: [   26.578823] eth0: 1000 Mbps Full duplex, port 0, queue 0
Wed Jun 20 13:42:45 2018 kern.info kernel: [   26.584222] br-lan: port 1(eth0) entered blocking state
Wed Jun 20 13:42:45 2018 kern.info kernel: [   26.589496] br-lan: port 1(eth0) entered forwarding state
Wed Jun 20 13:42:45 2018 daemon.notice netifd: Network device 'eth0' link is up
Wed Jun 20 13:42:45 2018 kern.info kernel: [   26.595495] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
Wed Jun 20 13:42:45 2018 daemon.notice netifd: bridge 'br-lan' link is up
Wed Jun 20 13:42:45 2018 daemon.notice netifd: Interface 'lan' has link connectivity

dnsmasq not working... (restarting service)

Wed Jun 20 13:51:49 2018 daemon.info dnsmasq[742]: exiting on receipt of SIGTERM
Wed Jun 20 13:51:49 2018 user.notice dnsmasq: DNS rebinding protection is active, will discard upstream RFC1918 responses!
Wed Jun 20 13:51:49 2018 user.notice dnsmasq: Allowing 127.0.0.0/8 responses
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: started, version 2.80test2 cachesize 150
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: DNS service limited to local subnets
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: compile time options: IPv6 GNU-getopt no-DBus no-i18n no-IDN DHCP no-DHCPv6 no-Lua TFTP no-conntrack no-ipset no-auth no-DNSSEC no-ID loop-detect inotify dumpfile
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq-dhcp[1227]: DHCP, IP range 192.168.1.100 -- 192.168.1.249, lease time 12h
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: using local addresses only for domain test
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: using local addresses only for domain onion
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: using local addresses only for domain localhost
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: using local addresses only for domain local
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: using local addresses only for domain invalid
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: using local addresses only for domain bind
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: using local addresses only for domain lan
Wed Jun 20 13:51:52 2018 daemon.warn dnsmasq[1227]: no servers found in /tmp/resolv.conf.auto, will retry
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: read /etc/hosts - 4 addresses
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq[1227]: read /tmp/hosts/dhcp.cfg01411c - 2 addresses
Wed Jun 20 13:51:52 2018 daemon.info dnsmasq-dhcp[1227]: read /etc/ethers - 0 addresses
Wed Jun 20 13:52:02 2018 daemon.info dnsmasq-dhcp[1227]: DHCPDISCOVER(br-lan) 192.168.1.211 00:00:00:00:00:00
...
```

dedeckeh commented:
Can you figure out what is causing this

Wed Jun 20 13:42:39 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Command failed: Not found
Wed Jun 20 13:42:39 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Failed to parse json data: unexpected end of data
Wed Jun 20 13:42:39 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Command failed: Not found
Wed Jun 20 13:42:39 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Failed to parse json data: unexpected end of data

Also you can verify if the procd interface trigger is in place by doing ubus call service list "{\"verbose\":true,\"name\":\"dnsmasq\"}

diizzyy commented:
Hmm... is there a way to set verbose loggning to see what's going on?
PROCD_DEBUG=1 doesn't seem to cut it, procd does have a verbose switch but I'm not sure how to enable that or if it'll help in this case at all?

diizzyy commented:
Not sure if this helps in any way

```
Wed Jun 20 14:21:40 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Command failed: Not found
Wed Jun 20 14:21:40 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Failed to parse json data: unexpected end of data
Wed Jun 20 14:21:40 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Command failed: Not found
Wed Jun 20 14:21:40 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: Failed to parse json data: unexpected end of data
Wed Jun 20 14:21:40 2018 daemon.notice procd: /etc/rc.d/S19dnsmasq: { "name": "dnsmasq", "script": "\/etc\/rc.d\/S19dnsmasq", "instances": { "cfg01411c": { "command": [ "\/usr\/sbin\/dnsmasq", "-C", "\/var\/etc\/dnsmasq.conf.cfg01411c", "-k", "-x", "\/var\/run\/dnsmasq\/dnsmasq.cfg01411c.pid" ], "file": [ "\/var\/etc\/dnsmasq.conf.cfg01411c" ], "respawn": [ "3600", "5", "5" ], "jail": { "name": "dnsmasq", "ubus": true, "log": true, "mount": { "\/var\/etc\/dnsmasq.conf.cfg01411c": "0", "\/usr\/share\/dnsmasq\/trust-anchors.conf": "0", "\/tmp\/hosts\/dhcp.cfg01411c": "0", "\/usr\/share\/dnsmasq\/rfc6761.conf": "0", "\/etc\/passwd": "0", "\/etc\/group": "0", "\/etc\/TZ": "0", "\/dev\/null": "0", "\/dev\/urandom": "0", "\/etc\/dnsmasq.conf": "0", "\/tmp\/dnsmasq.d": "0", "\/tmp\/resolv.conf.auto": "0", "\/etc\/hosts": "0", "\/etc\/ethers": "0", "\/sbin\/hotplug-call": "0", "\/usr\/lib\/dnsmasq\/dhcp-script.sh": "0", "\/var\/run\/dnsmasq\/": "1", "\/tmp\/dhcp.leases": "1" } } } }, "triggers": [ [ "config.change", [ "if", [ "eq", "p
Wed Jun 20 14:21:40 2018 daemon.info dnsmasq[745]: started, version 2.80test2 cachesize 150
```

dedeckeh commented:
Is this really the end of the json blob => "triggers": [ [ "config.change", [ "if", [ "eq", "p ?

diizzyy commented:
logread doesn't display any more than that at least

dedeckeh commented:
@diizzy Can you dump the output of ubus call service list "{"verbose":true,"name":"dnsmasq"}" after the device is booted ?

diizzyy commented:
I'm afriad it's to much of help, this time no error messages during boot (logread) but still not working until you restart dnsmasq.
```
root@OpenWrt:/# ubus call service list "{"verbose":true,"name":"dnsmasq"}"
Failed to parse message data
```
I get the same error msg even if I replace dnsmasq with dropbear for instance, also true on my WRT3200ACM.

diizzyy commented:
...going back to the official snapshot builds I can see the error, this is btw without WAN connected.

Running ubus call service list works however, no verbose info...

```
root@OpenWrt:/# ubus call service list
{
        "cron": {

        },
        "dnsmasq": {
                "instances": {
                        "cfg01411c": {
                                "running": true,
                                "pid": 1240,
                                "command": [
                                        "\/usr\/sbin\/dnsmasq",
                                        "-C",
                                        "\/var\/etc\/dnsmasq.conf.cfg01411c",
                                        "-k",
                                        "-x",
                                        "\/var\/run\/dnsmasq\/dnsmasq.cfg01411c.pid"
                                ],
                                "term_timeout": 5,
                                "respawn": {
                                        "threshold": 3600,
                                        "timeout": 5,
                                        "retry": 5
                                }
                        }
                }
```

dedeckeh commented:
You need to put escaping in place for the ubus call service list cmd
ubus call service list "{\\"verbose\\":true,\\"name\\":\\"dnsmasq\\"}"

diizzyy commented:
All information are from official snapshot builds except this post https://github.com/openwrt/openwrt/pull/1072#issuecomment-399249249
I'm back on official snapshot and with the errors during boot :-)

```
root@OpenWrt:/# ubus call service list "{\"verbose\":true,\"name\":\"dnsmasq\"}"
{
        "dnsmasq": {
                "instances": {
                        "cfg01411c": {
                                "running": true,
                                "pid": 1240,
                                "command": [
                                        "\/usr\/sbin\/dnsmasq",
                                        "-C",
                                        "\/var\/etc\/dnsmasq.conf.cfg01411c",
                                        "-k",
                                        "-x",
                                        "\/var\/run\/dnsmasq\/dnsmasq.cfg01411c.pid"
                                ],
                                "term_timeout": 5,
                                "respawn": {
                                        "threshold": 3600,
                                        "timeout": 5,
                                        "retry": 5
                                }
                        }
                },
                "triggers": [
                        [
                                "config.change",
                                [
                                        "if",
                                        [
                                                "eq",
                                                "package",
                                                "dhcp"
                                        ],
                                        [
                                                "run_script",
                                                "\/etc\/init.d\/dnsmasq",
                                                "reload"
                                        ]
                                ],
                                1000
                        ],
                        [
                                "config.change",
                                [
                                        "if",
                                        [
                                                "eq",
                                                "package",
                                                "system"
                                        ],
                                        [
                                                "run_script",
                                                "\/etc\/init.d\/dnsmasq",
                                                "reload"
                                        ]
                                ],
                                1000
                        ],
                        [
                                "interface.*",
                                [
                                        "if",
                                        [
                                                "eq",
                                                "interface",
                                                "lan"
                                        ],
                                        [
                                                "run_script",
                                                "\/etc\/init.d\/dnsmasq",
                                                "reload"
                                        ]
                                ],
                                1000
                        ]
                ]
        }
}
```

EricLuehrsen commented:
If interfaces are rapid-fire initializing, then dnsmasq can try to initialize in the middle of the noise. In `service_triggers()` change the procd delay variable which normally is 1000 [ms] `PROCD_RELOAD_DELAY=2000` and this has two benifits: the start is delayed after the last interface and if interfaces rapid-fire before the timeout then the timer restarts. This will change the JSON message for your service, but not affect any other service.

EricLuehrsen commented:
@diizzyy I assume this occurs when dnsmasq starts on one interface but another is coming up. Try to set `PROCD_RELOAD_DELAY=2000` in `service_triggers()` or any number above default 1000 [ms]. This has two benefits: (1) start is delayed after the last interface, and (2) timer resets for each interface that initializes before it expires. This affects the current service setup, but not all services. Worth a try anyway. Packages adblock and unbound use this method.

diizzyy commented:
@EricLuehrsen 
That sounds like a good idea! I'll give that a try. Thanks!

diizzyy commented:
@EricLuehrsen 
I gave that a try, didn't work :-/

diizzyy commented:
Hmm... seems like FreeBSD is doing something similar when using DHCP however I'm not sure if that's the actual race condition on OpenWrt.
https://svnweb.freebsd.org/base/release/11.2.0/etc/rc.d/defaultroute?revision=335563&view=markup

diizzyy commented:
Seeing this also on TP-Link WDR3600 using ath79, seems to be easier to trigger if there's no physical link on WAN.

psyborg55 commented:
disabling dnsmasq script reduced boot time for 3 seconds on my system. you might think again if this is acceptable solution. i'd say no.

gelstudios commented:
It seems like changing line 72 from `false) return 1;;` to `false) return 0;;` will help the situation without sleeping

diizzyy commented:
@gelstudios 
Out of curiosity, what hardware are you seeing this on?

jow- commented:
@diizzyy - can you please provide your exact `/etc/config/network` and `/etc/comnfig/dhcp` ?
